runtime/binding-mqtt/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1343,70 +1380,108 @@ private int decodePublishV5(
 
                 topicName = publish.topicName();
                 properties = publish.properties();
-                payload = publish.payload();
                 publishLimit = publish.limit();
             }
 
             final MqttPublishHeader mqttPublishHeader = mqttPublishHeaderRO.reset();
 
             reasonCode = mqttPublishHeader.decodeV5(server, topicName, properties, typeAndFlags, qos, packetId);
 
-            if (mqttPublishHeaderRO.payloadFormat.equals(MqttPayloadFormat.TEXT) && invalidUtf8(payload))
-            {
-                reasonCode = PAYLOAD_FORMAT_INVALID;
-                server.onDecodeError(traceId, authorization, reasonCode);
-                server.decoder = decodeIgnoreAll;
-                break decode;
-            }
-
             if (reasonCode == SUCCESS)
             {
-                final String topic = mqttPublishHeader.topic;
-                final long topicKey = topicKey(topic, qos);
+                final String topic = mqttPublishHeaderRO.topic;
+                final long topicKey = topicKey(topic, mqttPublishHeaderRO.qos);
                 MqttServer.MqttPublishStream publisher = server.publishes.get(topicKey);
 
                 if (publisher == null)
                 {
-                    publisher = server.resolvePublishStream(traceId, authorization, topic, qos);
+                    publisher = server.resolvePublishStream(traceId, authorization, topic, mqttPublishHeaderRO.qos);
```

Comment:
These should use `mqttPublishHeader` (local) instead of `mqttPublishHeader` (field), same instance, right?